ckan/controllers/user.py

```diff
@@ -80,7 +80,7 @@ def _setup_template_variables(self, context, data_dict):
         try:
             user_dict = get_action('user_show')(context, data_dict)
         except NotFound:
-            abort(404, _('User not found'))
+            abort(401, _('Authorize to see this page'))
```

Comment:
401 indicates an authentication error, not an authorization error, so it's not appropriate here.

If you want to send a redirect, send a redirect. Don't rely on repoze's magic redirection to a login when a 401 is sent.